psrc/ncglm.py

```diff
@@ -1,3 +1,32 @@
 # This is the Python wrapper for the ncglm library.
 
-# Maximilian McClelland 1/29/21
\ No newline at end of file
+# Maximilian McClelland 1/29/21
+
+import ctypes
+
+# Creating Libraries
+glm_group = ctypes.CDLL("../src/glm_group.so")
```

Comment:
OK, I'm starting to understand what you did here, and it's impressive. ;-) 

I see that what you want is to find the .so file for ncglm.

Let's start by taking all the Makefiles out of this PR, and also all the code changes where you added "include" to the include line, like this:
#include "../include/ncglm.h"

Leave the psrc and ptest changes.

I know this will break CI, but make those changes on your branch, and do a push, and then we'll go over the next step.